### PR TITLE
Fix documentation and update to Xcode 12

### PR DESCRIPTION
**Pull request by miguelangel-dev**
This PR fixes some pages in the documentation; these bugs have been introduced because nef had a bug [in the outcome reporter](https://github.com/bow-swift/nef/pull/178) and checked as "valid" some invalid pages in the documentation. 

As a reference, I attach [last CI log](https://pipelines.actions.githubusercontent.com/rPtncha3ApXa9wU6oGICHLlMNR1VplAuw39XHmEekrmgoIVCKo/_apis/pipelines/1/runs/487/signedlogcontent/3?urlExpires=2020-10-08T12%3A45%3A54.3592902Z&urlSigningMethod=HMACV1&urlSignature=UtF5rSz%2BQeHhVUU6X7ocDixRN2JbeiFHjkVuxRbtj38%3D) (check `Compile documentation` step)

> These changes will be together with `nef 0.7`; before this release, we have to do a double-check to documentation.

### CODE REVIEW

File: Documentation.app/Contents/MacOS/FP concepts.playground/Pages/Rank-N polymorphism.xcplaygroundpage/Contents.swift [hunk 1]
```diff
@@ -52,21 +56,25 @@ func singletonArray<T>(_ v: T) -> [T] {
...
+let letters = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ"
+// nef:end
 let n: Int = .random(in: 0 ..< 10)
+
 let randomWordGenerator: () -> String = { 
-    (0 ..< n).forEach { _ in 
-        .random(in: "a" ... "z") 
-    }
+    (0 ..< n).map { _ in
+        "\(letters.randomElement()!)"
```
Comment (truizlop) [inline Documentation.app/Contents/MacOS/FP concepts.playground/Pages/Rank-N polymorphism.xcplaygroundpage/Contents.swift:66]: For the sake of clarity, I'd maintain this as it is, I think it's easier to see the example.
```suggestion
        .random(in: "a" ... "z") 
```

File: Documentation.app/Contents/MacOS/FP concepts.playground/Pages/Rank-N polymorphism.xcplaygroundpage/Contents.swift [hunk 2]
```diff
@@ -52,21 +56,25 @@ func singletonArray<T>(_ v: T) -> [T] {
...
  */
 func makeRandomWordGenerator() -> () -> String {
     let length: Int = .random(in: 0 ..< 10)
-    return { 
-        (0 ..< length).forEach { _ in 
-            .random(in: "a" ... "z") 
-        }
+    return {
+        (0 ..< length).map { _ in
+            "\(letters.randomElement()!)"
```
Comment (truizlop) [inline Documentation.app/Contents/MacOS/FP concepts.playground/Pages/Rank-N polymorphism.xcplaygroundpage/Contents.swift:76]: Same here:
```suggestion
            .random(in: "a" ... "z") 
```

File: Documentation.app/Contents/MacOS/FP concepts.playground/Pages/Rank-N polymorphism.xcplaygroundpage/Contents.swift [hunk 3]
```diff
@@ -93,9 +101,9 @@ struct RandomWordGenerator {
     let length: Int
 
     func callAsFunction() -> String {
-        (0 ..< length).forEach { _ in 
-            .random(in: "a" ... "z")
-        }
+        (0 ..< length).map { _ in
+            "\(letters.randomElement()!)"
```
Comment (truizlop) [inline Documentation.app/Contents/MacOS/FP concepts.playground/Pages/Rank-N polymorphism.xcplaygroundpage/Contents.swift:105]: And here:
```suggestion
            .random(in: "a" ... "z") 
```

**Review by truizlop — CHANGES_REQUESTED** [review]

File: Documentation.app/Contents/MacOS/FP concepts.playground/Pages/Rank-N polymorphism.xcplaygroundpage/Contents.swift (hunk 1 above)
Comment (miguelangel-dev) [reply Documentation.app/Contents/MacOS/FP concepts.playground/Pages/Rank-N polymorphism.xcplaygroundpage/Contents.swift]: I have changed the approach, to keep the code as we had previously.

**Review by truizlop — APPROVED** [review]

**Pull request merged**